app/components/VaultControls/index.js

```diff
@@ -1045,10 +1081,15 @@ export default function VaultControls(props) {
                           (!approvalState.isApproved ||
                             approvalState.allowance === '0')
                         ) {
-                          setWithdrawLabel('Approve');
+                          // setWithdrawLabel('Approve');
+                        } else {
+                          dispatch(
+                            zapOutAllowance({
+                              allowance: true,
+                              vaultContract,
+                            }),
+                          );
                         }
```

Comment:
We should take all this logic away from the 'onChange' event, into the component mount since the approve state just depends on the allowance of the zapOut contract to spend the yvToken, independently of the selected zapOut target token. Also, since now the ETH vault displays by default a zap-out token (ETH), the 'onChange' event will not take place if user decides to zapOut without changing the target zapOut token.

Reply:
makes sense, moved it to useeffect